circleci: failed tests: TestChangeReplicasDescriptorInvariant

The following test appears to have failed:

[#20944](https://circleci.com/gh/cockroachdb/cockroach/20944):

```
I160802 22:03:31.336307 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:33330->127.0.0.1:44531: use of closed network connection
I160802 22:03:31.336743 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:39030->127.0.0.1:33574: use of closed network connection
I160802 22:03:31.339592 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
I160802 22:03:31.340394 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
I160802 22:03:31.341059 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- FAIL: TestChangeReplicasDescriptorInvariant (0.13s)
	client_raft_test.go:922: expected replica store=1:1 range=1 [/Min-/Max) to point to store 0, but got 2
=== RUN   TestProgressWithDownNode
I160802 22:03:31.343601 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
W160802 22:03:31.345266 gossip/gossip.go:1008  not connected to cluster; use --join to specify a connected node
I160802 22:03:31.355032 storage/replica.go:2607  gossiping cluster id {"00000000-0000-0000-0000-000000000000"} from store 1, range 1
I160802 22:03:31.356950 storage/replica.go:1990  store=1:1 range=1 [/Min-/Max): applying command with forced error: storage/replica.go:1911: no-op on empty Raft entry
E160802 22:03:31.357621 storage/replica.go:2025  store=1:1 range=1 [/Min-/Max): error executing raft command: storage/replica.go:1911: no-op on empty Raft entry
I160802 22:03:31.358294 storage/replica_trigger.go:308  store=1:1 range=1 [/Min-/Max): new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I160802 22:03:31.358350 storage/replica.go:2629  gossiping sentinel from store 1, range 1
I160802 22:03:31.358668 storage/replica.go:2636  gossiping first range from store 1, range 1: [{1 1 1}]
--
=== RUN   Example_rebalancing
W160802 22:04:15.286737 gossip/gossip.go:1008  not connected to cluster; use --join to specify a connected node
I160802 22:04:16.232605 stopper.go:408  quiesceing; tasks left:
19     gossip/infostore.go:290
--- PASS: Example_rebalancing (0.95s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	75.821s
=== RUN   TestBatchBasics
I160802 22:04:25.858547 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160802 22:04:25.861745 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.02s)
=== RUN   TestBatchRepr
I160802 22:04:25.882110 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160802 22:04:25.884002 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.02s)
=== RUN   TestBatchGet
I160802 22:04:25.898963 storage/engine/rocksdb.go:353  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -991,17 +991,6 @@ func TestChangeReplicasDescriptorInvariant(t *testing.T) {
 		t.Fatalf("got unexpected error: %v", err)
 	}
 
-	// Both addReplica calls attempted to use origDesc.NextReplicaID.
-	// The failed second call should not have overwritten the cached
-	// replica descriptor from the successful first call.
-	rep, err := mtc.stores[0].GetReplica(origDesc.RangeID)
-	if err != nil {
-		t.Fatalf("failed to look up replica %s: %s", origDesc.RangeID, err)
-	}
-	if a, e := rep.GetLastFromReplicaDesc().StoreID, mtc.stores[1].Ident.StoreID; a != e {
-		t.Fatalf("expected replica %s to point to store %s, but got %s", rep, a, e)
-	}
-
 	// Add to third store with fresh descriptor.
 	if err := addReplica(2, repl.Desc()); err != nil {
 		t.Fatal(err)

--- a/storage/helpers_test.go
+++ b/storage/helpers_test.go
@@ -164,9 +164,3 @@ func (r *Replica) GetTimestampCacheLowWater() hlc.Timestamp {
 	defer r.mu.Unlock()
 	return r.mu.tsCache.lowWater
 }
-
-func (r *Replica) GetLastFromReplicaDesc() roachpb.ReplicaDescriptor {
-	r.mu.Lock()
-	defer r.mu.Unlock()
-	return r.mu.lastFromReplica
-}

